docs/developer-guide/processing-log.rst

```diff
@@ -108,14 +108,24 @@ message.deserializationError.errorMessage (STRING)
 message.deserializationError.recordB64 (STRING)
   The Kafka record, encoded in Base64.
 
-message.recordProcessingError.errorMessage (STRING)
+message.recordProcessingError (STRUCT)
   The contents of a message with type 1 (RECORD_PROCESSING_ERROR). Logged when KSQL hits
   an error when processing a record, for example, an unexpected null value when evaluating
   an operator in a SELECT clause.
 
+message.recordProcessingError.errorMessage (STRING)
+  A string containing a human-readable error message detailing the error encountered.
+
 message.recordProcessingError.record (STRING)
   The KSQL record, serialized as a JSON string.
 
+message.productionError (STRUCT)
+  The contents of a message with type 2 (PRODUCTION_ERROR). Logged when a producer fails to
+  publish a Kafka record. These messages will only be logged if ``ksql.fail.on.production.error=false``.
```

Comment:
```suggestion
  publish a Kafka record. These messages are logged only if ``ksql.fail.on.production.error=false``.
```

Reply:
Whoops -- you actually caught a bug. This sentence was true in an earlier iteration of this PR but the most recent version logs error messages to the record processing log regardless of the value of the `ksql.fail.on.production.error` flag, so I've deleted the sentence.